broker-core/src/test/java/org/apache/qpid/server/security/group/FileGroupDatabaseCaseInsensitiveTest.java

```diff
@@ -360,6 +357,34 @@ public void testCreateGroupPersistedToFileCaseInsensitive() throws Exception
         assertTrue(newGroups.contains(MY_GROUP));
     }
 
+    @Test(expected = IllegalConfigurationException.class)
+    public void testDuplicateCreateGroupPersistedToFileCaseInsensitive() throws Exception
+    {
+        _util.writeAndSetGroupFile();
+
+        Set<String> groups = _fileGroupDatabase.getAllGroups();
+        assertTrue(groups.isEmpty());
+
+        _fileGroupDatabase.createGroup(MY_GROUP);
+
+        groups = _fileGroupDatabase.getAllGroups();
+        assertEquals(1, groups.size());
+        assertTrue(groups.contains(MY_GROUP));
+
+        _fileGroupDatabase.createGroup(MY_GROUP);
```

Comment:
the test will throw exception at this line

the rest of test code is "dead". please remove code after this line and please replace `MY_GROUP` with "MyGroup1"

Reply:
Removed the dead code, but replacing the group to mygroup1 wont result in the exception

Reply:
if group provider is case insensitive, the group "MyGroup1" would be a duplicate of "myGroup1". Thus, an exception would need to be thrown. Why it is not thrown?